Data/ByteString/Utils/ByteOrder.hs

```diff
@@ -0,0 +1,42 @@
+{-# LANGUAGE CPP #-}
+
+-- | Why does this module exist? There is "GHC.ByteOrder" in base.
+-- But that module is /broken/ until base-4.14/ghc-8.10, so we
+-- can't rely on it until we drop support for older ghcs.
+-- See https://gitlab.haskell.org/ghc/ghc/-/issues/20338
+-- and https://gitlab.haskell.org/ghc/ghc/-/issues/18445
+
+#include "MachDeps.h"
+
+module Data.ByteString.Utils.ByteOrder
+  ( ByteOrder(..)
+  , hostByteOrder
+  , whenLittleEndian
+  , whenBigEndian
+  ) where
+
+data ByteOrder
```

Comment:
Shall we reuse `GHC.ByteOrder.ByteOrder`?

Reply:
It makes sense to do so. It can wait until I make a pass at removing the various old-ghc cruft, though.

Reply:
I mean, we can already reuse the type itself?..

Reply:
It's added in `base-4.11` while we currently have `base >= 4.9`. I guess there is a compatibility package `ghc-byteorder` but that seems like re-arranging deck chairs on the Titanic since we will discontinue support for such old ghc versions at our convenience.

This isn't a battle worth fighting, though.